llvm/include/llvm/CodeGen/MachineOperand.h

```diff
@@ -446,6 +446,40 @@ class MachineOperand {
     return IsDebug;
   }
 
+  /// Whether this operand could be a sealed capability.
+  /// Note: this is very conservative and callers should look at the
+  /// instructions definition for register operands.
+  bool maybeSealed() const {
+    if (isReg())
+      return true; // Note: this is very conservative.
+    if (isFI())
+      return false; // FrameIndexes are always unsealed.
+    if (isCPI() || isTargetIndex() || isGlobal() || isBlockAddress() || isJTI())
+      return true; // Addresses of globals could be sealed.
```

Comment:
Only if function types? Even block addresses should be fine (at least currently).

Reply:
I'd rather be conservative here, targets can implement `TII:: isGuaranteedNotToTrap()` to override this behaviour.